cpmpy/solvers/pysat.py

```diff
@@ -243,100 +190,124 @@ def _after_solve(self, pysat_status):
                     cpm_var._value = None
                     pass
 
-        return self._solve_return(self.cpm_status)
-
+        return has_sol
 
-    def get_core(self):
+    def solver_var(self, cpm_var):
         """
-            For use with s.solve(assumptions=[...]). Only meaningful if the solver returned UNSAT. In that case, get_core() returns a small subset of assumption variables that are unsat together.
+            Transforms cpm_var into CNF literal using self.pysat_vpool
+            (positive or negative integer)
+        """
+        if isinstance(cpm_var, NegBoolView):
+            # just a view, get actual var identifier, return -id
+            pysat_var = -self.pysat_vpool.id(cpm_var._bv.name)
+        elif isinstance(cpm_var, _BoolVarImpl):
+            pysat_var = self.pysat_vpool.id(cpm_var.name)
+        else:
+            raise NotImplementedError(f"CPM_pysat: variable {cpm_var} not supported")
 
-            CPMpy will return only those variables that are False (in the UNSAT core)
+        # Add to varmap
+        self._varmap[cpm_var] = pysat_var
+        return pysat_var
 
-            Note that there is no guarantee that the core is minimal, though this interface does open up the possibility to add more advanced Minimal Unsatisfiabile Subset algorithms on top. All contributions welcome!
+    def _post_constraint(self, cpm_expr):
         """
-        assert hasattr(self, 'assumption_vars'), "get_core(): requires a list of assumption variables, e.g. s.solve(assumptions=[...])"
-        assert (self.cpm_status.exitstatus == ExitStatus.UNSATISFIABLE), "get_core(): solver must return UNSAT"
+            Post a primitive CPMpy constraint to the pysat solver API
+        """
+        from pysat.card import CardEnc
 
-        assum_idx = frozenset(self.pysat_solver.get_core()) # to speed up lookup
+        # base case, just var or ~var
+        if isinstance(cpm_expr, _BoolVarImpl):
+            self.pysat_solver.add_clause([self.solver_var(cpm_expr)])
+        elif isinstance(cpm_expr, Operator):
+            if cpm_expr.name == 'or':
+                self.pysat_solver.add_clause([self.solver_var(var) for var in cpm_expr.args])
+            else:
+                raise NotImplementedError(
+                    "Only 'or' operator supported by CPM_pysat for now (more possible with aiger, contact us on github")
```

Comment:
the reason of 'notimplemented' error is no longer true.
If there is some non-or constraint here then the 'to_cnf' does not support it...

so smth like f"Operator {cpm_expr} not automatically translated to CNF, please report on github"